### ISSUE TITLE
Export data generate a blank file 

### ISSUE THREAD

**Issue by uranio-235** [andOTP/andOTP#519]
After export data, result is a 0 bytes length file...

**Comment by flocke** [comment]
Do you export to any cloud storage or to your phones internal storage?

**Comment by docndev** [comment]
Here the same. It happens when I export to my internal storage.

**Comment by flocke** [comment]
Which export are you using? Password protected, PGP, Plain-text?

**Comment by guyfw** [comment]
I tried both with plain and password-protected. I noticed that this could be solved waiting approximately 15s after the beginning of the export procedure.

**Comment by beerisgood** [comment]
15s? Sound you try to save the file to a slow sd card or you use a "security" solution
It's only few Kb.

**Comment by guyfw** [comment]
Yeah, I saved on my SD card (60 MB/s in writing) since it's the only path available to me.

**Comment by beerisgood** [comment]
> Yeah, I saved on my SD card (60 MB/s in writing) since it's the only path available to me.

Why? Every phone has internal storage.
Also are you sure that your sd card isn't corrupt? With 60MB/s you wouldn't get any trouble

**Comment by guyfw** [comment]
I'm sorry, I though the Download folder was located on my SD. I tried to save on it and I had no issues. However, I still do not comprehend why there's such difference since I checked my SD for integrity without reporting errors.

**Comment by alexbakker** [comment]
We're getting similar reports: https://github.com/beemdevelopment/Aegis/issues/362. It appears that no exception is thrown when writing the file, but it ends up being 0 bytes in size somehow. Could it be that the Storage Access Framework is buggy on some devices?

**Comment by beerisgood** [comment]
> We're getting similar reports: [beemdevelopment/Aegis#362](https://github.com/beemdevelopment/Aegis/issues/362). It appears that no exception is thrown when writing the file, but it ends up being 0 bytes in size somehow. Could it be that the Storage Access Framework is buggy on some devices?

I would say the error is on app side. Also with upcomming Android 11 the actually storage permission get changed, so apps now need to change that as well.
Read: https://www.reddit.com/r/GrapheneOS/comments/fre3vz/storage_permissions/flvqzhi/

**Comment by alexbakker** [comment]
> I would say the error is on app side.

What makes you draw this conclusion? The (little) evidence we have so far suggests otherwise. I hope to be proven wrong though.

> Also with upcomming Android 11 the actually storage permission get changed, so apps now need to change that as well.

Indeed. Aegis already doesn't request the legacy storage permissions anymore.

**Comment by alexbakker** [comment]
FYI: One of our users reported [the following](https://github.com/beemdevelopment/Aegis/issues/362#issuecomment-624097631):

> the issue is solved, it was not caused of aegis. had the same problem just now when exporting my contacts and was also a 0 Byte empty file, so i searched in settings and found in the developer options the point "allow external storage usage of apps" since activating , everything works as desired.

**Comment by flocke** [comment]
> FYI: One of our users reported [the following](https://github.com/beemdevelopment/Aegis/issues/362#issuecomment-624097631): ...

I will put this issue on hold until after the storage rework that is planed for version 0.8.0, which should fix most of the storage-related problems.

**Comment by skobkin** [comment]
Same here with GPG (OpenKeychain) export.

**Comment by flocke** [comment]
Please try again with version 0.8.0, the entire storage access was reworked.

**Comment by skobkin** [comment]
> Please try again with version 0.8.0

It was report for 0.8.0. But I tried GPG backup again just in case and it resulted with the same 0 byte file being created in the backup directory. Didn't try other backup types again because I hadn't problem with them.

**Comment by flocke** [comment]
I'm moving this to the next milestone because I can not find a way to successfully reproduce this so I can't really debug anything here. I also haven't had the issue in a while now (using GPG with Nextcloud as backup location).
If this still happens after the release of 0.9.0 I will have to look into it again, maybe the solution from https://github.com/beemdevelopment/Aegis/issues/362 can help here as well.

**Comment by Avamander** [comment]
I just encountered this as well with OpenPGP export on Android 11.
